service/src/main/kotlin/fi/espoo/evaka/attendance/ChildAttendanceController.kt

```diff
@@ -380,6 +376,40 @@ class ChildAttendanceController(
             tx.getAttendancesResponse(unitId)
         }.let { ResponseEntity.ok(it) }
     }
+
+    data class AbsenceRangeDeleteRequest(
+        val period: FiniteDateRange
+    )
+    @PostMapping("/units/{unitId}/children/{childId}/absence-range/delete")
+    fun deleteAbsenceRange(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        @PathVariable childId: UUID,
+        @RequestBody body: AbsenceRangeDeleteRequest
+    ): ResponseEntity<Unit> {
+        Audit.AbsenceDeleteRange.log(targetId = childId)
+        acl.getRolesForChild(user, childId).requireOneOfRoles(UserRole.MOBILE)
+        return db.transaction { tx ->
+            tx.deleteAbsencesByPeriod(childId, body.period)
+        }.let { ResponseEntity.ok(it) }
+    }
+
+    data class AbsenceDateDeleteRequest(
+        val date: LocalDate
+    )
+    @PostMapping("/units/{unitId}/children/{childId}/absence")
+    fun deleteAbsence(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        @PathVariable childId: UUID,
+        @RequestBody body: AbsenceDateDeleteRequest
+    ): ResponseEntity<Unit> {
+        Audit.AbsenceDelete.log(targetId = childId)
+        acl.getRolesForChild(user, childId).requireOneOfRoles(UserRole.MOBILE)
+        return db.transaction { tx ->
+            tx.deleteAbsencesByDate(childId, body.date)
+        }.let { ResponseEntity.ok(it) }
+    }
```

Comment:
The latter endpoint does not even mention delete in path so it's very surprising that such an endpoint deletes something. I would change these from post to delete and pass date/range as request param to keep it more restfull.